bee-tangle/src/ms_tangle.rs

```diff
@@ -338,6 +348,16 @@ impl<B: StorageBackend> MsTangle<B> {
         self.solid_entry_points.lock().await.clear();
     }
 
+    /// Replaces all solid entry points.
+    pub async fn replace_solid_entry_points(
+        &self,
+        new_seps: impl IntoIterator<Item = (SolidEntryPoint, MilestoneIndex)>,
```

Comment:
I don't know how important is the fact that this is an iterator. But if it were to be an slice. you could use `seps.copy_from_slice(new_seps)` (which is a `memcpy`) and then `seps.truncate(new_seps.len())`.

Reply:
okay, good point, let me check

Reply:
`seps` is not a slice. It's currently a HashMap.

Reply:
ok then :)